### PR TITLE
Refactor Database folder

### PR DESCRIPTION
**Pull request by sasha370**
We plan to use PostgreSQL as a DB instead of SQLite and file storage.
We need to extract DB engine and session controller to a separate class.
We want to extract all models as well. 

Additional: 
- refactoring an clean up

### How to test
- manual tests with instruction  https://gist.github.com/sasha370/1c218f78bd0d2b2e41289b4814e0c778

### CODE REVIEW

File: database/interaction_manager.py [hunk 1]
```diff
@@ -1,161 +1,111 @@
...
+            print(f"Error getting interactions by interaction IDs: {e}")
 
     def get_qa_interactions(self):
-        session = self.Session()
         try:
-            return session.query(QAInteractions).all()
-        finally:
-            session.close()
-
-    def get_all_interactions(self):
```
Comment (sasha370) [inline database/interaction_manager.py:108]: Dubled :arrow_up:   `get_qa_interactions`

File: database/database.py [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+            cls._instance = super().__new__(cls)
+            cls._instance._init_engine()
+            cls._instance._create_tables()
+        return cls._instance
+
+    def _create_tables(self):
+        """
+        Create tables in the database if they do not exist.
+        """
+        for model in [QAInteractions, SpaceInfo, PageData, BookmarkedConversation, QuizQuestion, UserScore]:
```
Comment (sasha370) [inline database/database.py:46]: We need to know about all models to be able to create tables during first initialization of Database

File: database/page_manager.py [hunk 3]
```diff
@@ -1,284 +1,153 @@
...
-            f"Page id: {record[1]}, space key: {record[2]}, title: {record[3]}, "
-            f"author: {record[4]}, created date: {record[5]}, last updated: {record[6]}, "
-            f"content: {record[7]}, comments: {record[8]}"
-        )
-        all_documents.append(document)
-        retrieved_page_ids.append(record[1])
-
-    # Close the SQLite connection
-    conn.close()
-    return all_documents, retrieved_page_ids
```
Comment (sasha370) [inline database/page_manager.py:211]: Removed. Not used

File: database/page_manager.py [hunk 4]
```diff
@@ -1,284 +1,153 @@
...
-    :return:
-    """
-    conn = sqlite3.connect(sql_file_path)
-    cursor = conn.cursor()
-    current_time = datetime.now()
-    for page_id in page_ids:
-        cursor.execute("UPDATE page_data SET last_embedded = ? WHERE page_id = ?", (current_time, page_id))
-    conn.commit()
-    conn.close()
-    return True
```
Comment (sasha370) [inline database/page_manager.py:227]: Removed. Not used

File: database/quiz_question_manager.py [hunk 5]
```diff
@@ -1,38 +1,20 @@
...
-
-
-class QuizQuestion(Base):
-    __tablename__ = 'quiz_questions'
-    id = Column(Integer, primary_key=True)
-    question_text = Column(Text)
-    thread_id = Column(String)  # Slack thread ID for tracking conversations
-    summary = Column(Text)  # Summary of the conversation
-    posted_on_slack = Column(DateTime)  # Timestamp when posted on Slack
-    posted_on_confluence = Column(DateTime, nullable=True)  # Timestamp when posted on Confluence
```
Comment (sasha370) [inline database/quiz_question_manager.py:19]: Extracted

File: database/quiz_question_manager.py [hunk 6]
```diff
@@ -78,20 +60,9 @@ def update_with_thread_id(self, question_id, thread_id):
...
-    def update_confluence_timestamp(self, question_id):
-        try:
-            with self.Session() as session:
-                question = session.query(QuizQuestion).filter_by(id=question_id).first()
-                if question:
-                    question.posted_on_confluence = datetime.now(timezone.utc)
-                    session.commit()
-        except SQLAlchemyError as e:
-            print(f"Error updating Confluence timestamp: {e}")
-
```
Comment (sasha370) [inline database/quiz_question_manager.py:90]: Removed. Not used

File: database/space_manager.py [hunk 7]
```diff
@@ -1,35 +1,11 @@
...
-class SpaceInfo(Base):
-    """
-    SQLAlchemy model for storing Confluence space data.
-    """
-    __tablename__ = 'space_info'
-
-    id = Column(Integer, primary_key=True)
-    space_key = Column(String, nullable=False)
-    space_name = Column(String, nullable=False)
-    last_import_date = Column(DateTime, nullable=False)
```
Comment (sasha370) [inline database/space_manager.py:21]: Extracted

File: models/qa_interaction.py [hunk 8]
```diff
@@ -0,0 +1,19 @@
+from .base import *
+import json
+
+
+class QAInteractions(Base):
```
Comment (sasha370) [inline models/qa_interaction.py:5]: I'd like to rename it to `QAInteraction` but not sure about it. 
Do we plan to launch this version on the existing DB?

File: configuration.py [hunk 9]
```diff
@@ -22,6 +22,7 @@ def get_project_root() -> str:
 
 chart_folder_path = os.path.join(project_path, "content", "charts")
 sql_file_path = os.path.join(project_path, "content", "database", "confluence_pages_sql.db")
+db_url = 'sqlite:///' + sql_file_path
```
Comment (sasha370) [inline configuration.py:25]: We can use `db_url = 'postgresql://username:password@localhost/db_name'` for psql

File: database/database.py (hunk 2 above)
Comment (oleg-polivannyi) [reply database/database.py:46]: If we are going to have more models, we may have an index file, or better a file, that will scan its directory and provide an array of imported models

**Review by oleg-polivannyi — APPROVED** [review]

File: models/qa_interaction.py [hunk 10]
```diff
@@ -0,0 +1,19 @@
+from .base import *
+import json
+
+
+class QAInteraction(Base):
```
Comment (sasha370) [inline models/qa_interaction.py:5]: renamed from `QAInteractions`  to `QAInteraction`

File: models/qa_interaction.py [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    __tablename__ = 'qa_interactions'
+
+    id = Column(Integer, primary_key=True)
```
Comment (sasha370) [inline models/qa_interaction.py:8]: Renamed from `interaction_id` to `id`

File: slack/event_consumer.py [hunk 12]
```diff
@@ -63,13 +63,13 @@ def add_question_and_response_to_database(self, question_event, response_text, a
     def process_question(self, question_event: QuestionEvent):
         channel_id = question_event.channel
         message_ts = question_event.ts
-        user_id = question_event.user
```
Comment (sasha370) [inline slack/event_consumer.py:66]: Not used

File: database/database.py [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+        """
+        Rollback the current transaction.
+
+        This method rolls back the current transaction in the active session.
+
+        Raises:
+            SQLAlchemyError: If an error occurs while rolling back the transaction.
+        """
+        try:
+            self.get_session().rollback()
```
Comment (id-ilych) [inline database/database.py:101]: Why is rollback performed on a new session?

File: database/database.py [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+from configuration import db_url
+from models.qa_interaction import QAInteractions
+from models.space_info import SpaceInfo
+from models.page_data import PageData
+from models.bookmarked_conversation import BookmarkedConversation
+from models.quiz_question import QuizQuestion
+from gamification.models.user_score import UserScore
+
+
+class Database:
```
Comment (id-ilych) [inline database/database.py:13]: Is there a point in singleton class, isn't the module a singleton already?

File: gamification/models/user_score.py
Comment (id-ilych) [inline gamification/models/user_score.py:1]: Please consider moving gamification to database / models, if it makes sense.

**Review by id-ilych — APPROVED** [review]
Is there a benefit in `*Manager` classes as opposed to module methods, that receive db connection (or session/transaction)?

File: database/database.py (hunk 13 above)
Comment (sasha370) [reply database/database.py:101]: You were right about a new session. I remove this method and will call rollback() on the session itself

File: database/database.py (hunk 14 above)
Comment (sasha370) [reply database/database.py:13]: I will do it in the next ticket

**Pull request merged**